File: instagram-frontend-mobile/services/post-like.service.ts
import axiosInstance from "../config/axiosInstance";
import { API_ENDPOINTS } from "../config/routes";
import { ApiResponse, PaginatedResponse } from "../types";
import { PostLikeToggleResponse, PostLikeUserResponse } from "../types/post.type";

export const postLikeService = {
    toggleLikePost: async (postId: string): Promise<PostLikeToggleResponse> => {
        try {
            const response = await axiosInstance.post(API_ENDPOINTS.TOGGLE_LIKE_POST(postId));
            return response.data;
        } catch (error: any) {
            console.error('[postLikeService] Error details:', {
                postId,
                status: error.response?.status,
                statusText: error.response?.statusText,
                data: error.response?.data,
                message: error.message,
            });
            const errorMsg = error.response?.data?.message || error.message || 'Failed to toggle like';
            throw new Error(errorMsg);
        }
    },

    getPostLikes: async (postId: string, page = 0, pageSize = 20, query = ''): Promise<PaginatedResponse<PostLikeUserResponse>> => {
        try {
            const response = await axiosInstance.get<ApiResponse<PaginatedResponse<PostLikeUserResponse>>>(
                API_ENDPOINTS.GET_POST_LIKES(postId),
                { params: { page, pageSize, query } }
            );
            return response.data.data;
        } catch (error: any) {
            console.error('[postLikeService] Failed to get post likes:', error);
            throw new Error(error.response?.data?.message || 'Failed to load likes');
        }
    },
};

File: instagram-backend/src/main/java/com/hoanghuy04/instagrambackend/service/post/PostLikeService.java
package com.hoanghuy04.instagrambackend.service.post;

import com.hoanghuy04.instagrambackend.dto.response.PageResponse;
import com.hoanghuy04.instagrambackend.dto.response.PostLikeToggleResponse;
import com.hoanghuy04.instagrambackend.dto.response.PostLikeUserResponse;
import org.springframework.data.domain.Pageable;


public interface PostLikeService {
    PostLikeToggleResponse toggleLikePost(String postId);

    PageResponse<PostLikeUserResponse> getPostLikes(String postId, String query, Pageable pageable);
}
